### ISSUE TITLE
Assert in Words game

### ISSUE THREAD

**Issue by mmandlis** [PolymerLabs/arcs#1865]
Happens on arcs-live every second move or so:

assert-web.js:11 Uncaught (in promise) Error: cannot add without new keys
    at assert (assert-web.js:11)
    at CrdtCollectionModel.add (crdt-collection-model.js:45)
    at CollectionProxy._processUpdate (storage-proxy.js:267)
    at CollectionProxy._processUpdates (storage-proxy.js:185)
    at CollectionProxy._onUpdate (storage-proxy.js:150)
    at callback (storage-proxy.js:83)
    at PECInnerPort.ParticleExecutionContext._apiPort.onSimpleCallback (particle-execution-context.js:67)
    at PECInnerPort._handle (api-channel.js:174)
    at MessagePort.APIPort._port.onmessage (api-channel.js:90)

**Comment by mmandlis** [comment]
Related or not, 
Every move I make in the game, creates a new Board under `backingStores` in firebase, and the version is forwarded by 2 or more. 
Deleting the arc doesn't delete any of the boards.

**Comment by shans** [comment]
Deleting the arc shouldn't delete the data. It sounds like the Board has a new id every time it's written? That seems maybe wrong - we should eventually be relying on entity mutation I would think, so should simulate that for now by writing with the same id.

'cannot add without new keys' - any chance you can figure out what data is being written when this triggers? Is it the board?

**Comment by mmandlis** [comment]
The data that's being written has `renderRecipe` and `renderParticleSpec`, so i think it's a `Post` (https://github.com/PolymerLabs/arcs/blob/master/artifacts/Social/Post.schema#L27) .

Also I noticed some inconsistency - depending when i leave the arc with the words game and go back to launcher, it is or isn't reloaded properly when I open it again (even it's name is sometimes not set to "play words"). The problem is reproducible on arcs-live, but I haven't spend too much time investigating it.

Btw, is there a way now for the data to be deleted?

**Comment by shans** [comment]
No, the idea is that data remains present because it might be used by more than one arc (before this wasn't a problem as the data was always copied). We might look at a way to refcount data so that we can clean up genuinely unused entities in the future.
